ouroboros-consensus-cardano-test/test/Test/Consensus/Cardano/Translation.hs

```diff
@@ -0,0 +1,226 @@
+import           Test.Tasty.QuickCheck
+
+import qualified Cardano.Chain.Block as Byron
+import qualified Cardano.Chain.UTxO as Byron
+import           Cardano.Ledger.BaseTypes (Network (Testnet), TxIx (..))
+import qualified Cardano.Ledger.Era as Core
+import           Cardano.Ledger.Shelley.API (ShelleyGenesis (..),
+                     ShelleyGenesisStaking (..), TxIn (..), TxOut,
+                     translateCompactTxOutByronToShelley,
+                     translateTxIdByronToShelley)
+import           Cardano.Ledger.Shelley.PParams (emptyPParams)
+import           Cardano.Slotting.EpochInfo (fixedEpochInfo)
+import           Cardano.Slotting.Slot (EpochNo (..))
+
+import           Ouroboros.Consensus.BlockchainTime.WallClock.Types (SlotLength,
+                     slotLengthFromSec)
+import           Ouroboros.Consensus.Byron.Ledger (ByronBlock, byronLedgerState)
+import           Ouroboros.Consensus.Cardano.Block (CardanoEras, ShelleyEra)
+import           Ouroboros.Consensus.Cardano.CanHardFork ()
+import           Ouroboros.Consensus.HardFork.Combinator (EraTranslation,
+                     InPairs (..), hardForkEraTranslation, translateLedgerState)
+import           Ouroboros.Consensus.HardFork.Combinator.State.Types
+                     (TranslateLedgerState (translateLedgerStateWith))
+import           Ouroboros.Consensus.HardFork.Combinator.Util.InPairs
+                     (provideBoth)
+import           Ouroboros.Consensus.Ledger.Basics (ApplyMapKind' (..), DiffMK,
+                     EmptyMK, LedgerConfig, LedgerState)
+import           Ouroboros.Consensus.Protocol.Praos.Common
+                     (MaxMajorProtVer (..))
+import           Ouroboros.Consensus.Protocol.Praos.Translate ()
+import           Ouroboros.Consensus.Protocol.TPraos (TPraos)
+import           Ouroboros.Consensus.Shelley.Eras
+import           Ouroboros.Consensus.Shelley.HFEras ()
+import           Ouroboros.Consensus.Shelley.Ledger (ShelleyBlock,
+                     ShelleyLedgerConfig, mkShelleyLedgerConfig,
+                     shelleyLedgerTables, shelleyUTxOTable)
+import           Ouroboros.Consensus.Shelley.Ledger.SupportsProtocol ()
+import           Ouroboros.Consensus.Storage.LedgerDB.HD (UtxoDiff (..),
+                     UtxoEntryDiff (..), UtxoEntryDiffState (..))
+import           Ouroboros.Consensus.TypeFamilyWrappers
+
+import           Test.Cardano.Ledger.Shelley.Utils (unsafeBoundRational)
+import           Test.Consensus.Byron.Generators (genByronLedgerConfig,
+                     genByronLedgerState)
+import           Test.Consensus.Cardano.MockCrypto (MockCryptoCompatByron)
+import           Test.Consensus.Shelley.Generators ()
+import           Test.Consensus.Shelley.MockCrypto
+import           Test.Util.Time (dawnOfTime)
+
+import qualified Cardano.Ledger.Block as Core
+import qualified Cardano.Ledger.Core as Core
+import qualified Cardano.Ledger.Era as Core
+
+type Crypto = MockCryptoCompatByron
```

Comment:
I wonder if we should remove this level of indirection.

Reply:
I think it is unnecessary indirection, but it reduces noise in type signatures. I am impartial to this one.

Reply:
I'd prefer skipping the extra level of indirection, but that's subjective. @Jasagredo preferences?

Reply:
I agree with both, but I also tend to read files as @Jasagredo. So just to close the discussion I am keeping it as it is.